### PR TITLE
RHIDP-777 [Test automation] Verify Tekton is a supported plugin

### PR DESCRIPTION
**Pull request by gustavolira**
## Description

Please explain the changes you made here.

## Which issue(s) does this PR fix

https://issues.redhat.com/browse/RHIDP-777

- Fixes #?
Relates to Jira: RHIDP-777

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: e2e-tests/playwright/utils/tekton/tekton.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    await this.page
+      .locator(
+        'table[aria-labelledby="Pipeline Runs"] button[aria-label="expand row"]',
+      )
+      .first()
+      .click();
+  }
+
+  async openModalEchoHelloWorld() {
+    const loc = 'g[data-test="task echo-hello-world"]';
```
Comment (subhashkhileri) [inline e2e-tests/playwright/utils/tekton/tekton.ts:27]: can we move this locator to pageObjects.

File: e2e-tests/playwright/utils/tekton/tekton.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    await this.page.waitForSelector(loc);
+    await this.page.locator(loc).click();
+  }
+
+  async verifyTableHasRows() {
+    const rowCount = await this.page
+      .locator('table[aria-labelledby="Pipeline Runs"] tbody tr')
```
Comment (subhashkhileri) [inline e2e-tests/playwright/utils/tekton/tekton.ts:34]: can we move this locator to pageObjects.

File: e2e-tests/playwright/utils/tekton/tekton.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      .count();
+    expect(rowCount).toBeGreaterThan(0);
+  }
+
+  async search(value: string) {
+    await this.page.getByPlaceholder('Search').fill(value);
+  }
+
+  async isModalOpened() {
+    await expect(this.page.locator('#pipelinerun-logs')).toBeVisible();
```
Comment (subhashkhileri) [inline e2e-tests/playwright/utils/tekton/tekton.ts:44]: can we move this locator to pageObjects.

File: e2e-tests/playwright/utils/tekton/tekton.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  async goToBackstageJanusProject() {
+    await this.uiHelper.openSidebar('Catalog');
+    await this.uiHelper.clickLink('backstage-janus');
+    await this.uiHelper.clickTab('CI');
+    await this.page.waitForSelector('h2:text("Pipeline Runs")');
```
Comment (subhashkhileri) [inline e2e-tests/playwright/utils/tekton/tekton.ts:51]: I think we can use uiHelper.verifyHeading() for this, just a suggestion.

File: e2e-tests/playwright/utils/tekton/tekton.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  async checkPipelineStages(texts: string[]) {
+    for (const text of texts) {
+      const stage = this.page.locator(`h6:text("${text}")`);
```
Comment (subhashkhileri) [inline e2e-tests/playwright/utils/tekton/tekton.ts:56]: I think we can use uiHelper.verifyHeading() for this as well.

File: e2e-tests/playwright/utils/tekton/tekton.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+      await expect(stage).toBeVisible();
+    }
+  }
+
+  async checkPipelineOutput(texts: string[]) {
+    for (const text of texts) {
+      const locator = this.page.locator(`span:text("${text}")`);
```
Comment (subhashkhileri) [inline e2e-tests/playwright/utils/tekton/tekton.ts:63]: can we use `uihelper.verifyText()` for text verification.

File: e2e-tests/playwright/e2e/plugins/tekton/tekton.spec.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    const page = await browser.newPage();
+    common = new Common(page);
+    await common.loginAsGuest();
+    uiHelper = new UIhelper(page);
+    tekton = new Tekton(page);
+  });
+
+  test('Check Pipeline Run', async () => {
+    await tekton.goToBackstageJanusProject();
+    await tekton.verifyTableHasRows();
```
Comment (subhashkhileri) [inline e2e-tests/playwright/e2e/plugins/tekton/tekton.spec.ts:27]: verifyTableHasRows seems a bit confusing as to which table it is verifying. Can we rename it or pass pipeline runs as a parameter?

File: e2e-tests/playwright/utils/tekton/tekton.ts (hunk 2 above)
Comment (gustavolira) [reply e2e-tests/playwright/utils/tekton/tekton.ts:34]: @subhashkhileri this locator is very specific for this page

File: e2e-tests/playwright/utils/tekton/tekton.ts (hunk 3 above)
Comment (gustavolira) [reply e2e-tests/playwright/utils/tekton/tekton.ts:44]: @subhashkhileri this locator is very specific for this page

File: e2e-tests/playwright/utils/tekton/tekton.ts (hunk 1 above)
Comment (gustavolira) [reply e2e-tests/playwright/utils/tekton/tekton.ts:27]: @subhashkhileri this locator is very specific for this page

File: .ibm/pipelines/value_files/values_showcase.yaml [hunk 8]
```diff
@@ -43,6 +40,24 @@ global:
           disabled: false
         - package: ./dynamic-plugins/dist/janus-idp-backstage-plugin-keycloak-backend-dynamic
           disabled: false
+        - package: ./dynamic-plugins/dist/janus-idp-backstage-plugin-tekton
+          disabled: false
```
Comment (invincibleJai) [inline .ibm/pipelines/value_files/values_showcase.yaml:44]:
```suggestion
        - package: ./dynamic-plugins/dist/janus-idp-backstage-plugin-tekton
          disabled: false
        - package: ./dynamic-plugins/dist/janus-idp-backstage-plugin-tekton
          disabled: true
          pluginConfig:
            dynamicPlugins:
              frontend:
                janus-idp.backstage-plugin-tekton:
                  mountPoints:
                    - mountPoint: entity.page.ci/cards
                      importName: TektonCI
                      config:
                        layout:
                          gridColumn: "1 / -1"
                        if:
                          allOf:
                            - isTektonCIAvailable
```

do you need this as well?

File: dynamic-plugins.default.yaml [hunk 9]
```diff
@@ -653,20 +653,7 @@ plugins:
...
-          janus-idp.backstage-plugin-tekton:
-            mountPoints:
-              - mountPoint: entity.page.ci/cards
-                importName: TektonCI
-                config:
-                  layout:
-                    gridColumn: "1 / -1"
-                  if:
-                    allOf:
-                      - isTektonCIAvailable
```
Comment (invincibleJai) [inline dynamic-plugins.default.yaml:669]: why remove this?

**Pull request merged**